### ISSUE TITLE
JPlag viewer displaying strange characters

### ISSUE THREAD

**Issue by muety** [jplag/JPlag#2110]
**Original code:** 

```java
// [...]
		    /**
		     * Gibt eine einzelne Ziffer einer Zahl anhand eines Index zurueck.
		     * 
		     * @param number
		     * @param index
		     * @return Die Ziffer von der Zahl 'number' an dem Index 'index'.
		     */
		    public static int digitAt(int number, int index) {
		        return String.valueOf(number).charAt(index) - '0';
		    


	}

}
```

**JPlag viewer:**
![image001](https://github.com/user-attachments/assets/f577bebe-c200-45ff-9dc6-a4228672c4fb)

Note the wrong characters in line 28. The `.java` file is of UTF-8 encoding. When opening it in a text editor (with UTF-8 being correctly recognized as the encoding), things look normal.

**Comment by Kr0nox** [comment]
@muety Could you attach the file to the issue, so we can reproduce it easier and find the error

**Comment by tsaglam** [comment]
This might be a conversion issue. If the student's file is not UTF-8, we try to guess the file encoding and convert it to UTF-8 in order to prevent issues where the Java compiler cannot parse the file. However, there is inherently no way to always correctly determine the file encoding if it is unknown. Thus, sometimes it is incorrectly guessed. Note: this issue can also arise if you convert the files yourself before submitting them to JPlag. Some learning platforms, such as Artemis, might also do this.

**Comment by muety** [comment]
I can't share the file, unfortunately, because I'd first have to trim sensitive information, which would potentially also modify encoding, etc.

When downloading the file from ILIAS learning platform,  it has UTF-8 encoding and everything looks fine regardless of which editor I open it in. Only JPlag will show those weird characters.

Same for many other students (mostly using Eclipse on Windows).

**Comment by Kr0nox** [comment]
This is fixed in #2118
